### PR TITLE
 test(ssr): fix SSR flakes and change matchMedia variable name

### PR DESCRIPTION
**Pull request by ThomasBurleson**
* improve use of inject() in tests

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
Plz review @CaerusKaru

File: src/lib/core/match-media/match-media.spec.ts [hunk 1]
```diff
@@ -23,16 +25,24 @@ describe('match-media', () => {
...
+      (_mediaObserver: MediaObserver, _matchMedia: MockMatchMedia,
+       _breakPoints: BreakPointRegistry) => {
+        breakPoints = _breakPoints;
+        mediaController = _matchMedia;      // inject only to manually activate mediaQuery ranges
+        mediaObserver = _mediaObserver;
+      }));
+
   afterEach(() => {
-    matchMedia.clearAll();
+    if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/match-media/match-media.spec.ts]: Do we need this?

File: src/lib/core/media-observer/media-observer.spec.ts [hunk 2]
```diff
@@ -18,158 +18,166 @@ import {MockMatchMedia, MockMatchMediaProvider} from '../match-media/mock/mock-m
...
 
-        } finally {
-          matchMedia.autoRegisterQueries = true;
-          subscription.unsubscribe();
+      } finally {
+        mediaController.autoRegisterQueries = true;
+        subscription.unsubscribe();
 
-          matchMedia.clearAll();
+        if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/media-observer/media-observer.spec.ts]: Do we need this here?

File: src/lib/core/media-observer/media-observer.spec.ts [hunk 3]
```diff
@@ -18,158 +18,166 @@ import {MockMatchMedia, MockMatchMediaProvider} from '../match-media/mock/mock-m
...
+      expect(current.mqAlias).toBe('md');
 
-        matchMedia.activate('xs');
-        expect(current.mqAlias).toBe('md');
+      mediaController.activate('xs');
+      expect(current.mqAlias).toBe('md');
 
-        matchMedia.clearAll();
-      }));
+      if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/media-observer/media-observer.spec.ts]: Do we need this here?

File: src/lib/core/media-observer/media-observer.spec.ts [hunk 4]
```diff
@@ -18,158 +18,166 @@ import {MockMatchMedia, MockMatchMediaProvider} from '../match-media/mock/mock-m
...
+      subscription.unsubscribe();
 
-        matchMedia.activate('lg');
-        expect(current.mqAlias).toBe('xs');
+      mediaController.activate('lg');
+      expect(current.mqAlias).toBe('xs');
 
-        matchMedia.clearAll();
-      }));
+      if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/media-observer/media-observer.spec.ts]: Do we need this here?

File: src/lib/core/media-observer/media-observer.spec.ts [hunk 5]
```diff
@@ -192,117 +200,146 @@ describe('media-observer', () => {
...
+      // unless a printAlias layout has been configured.
+      mediaController.activate('slate.xl');
+      expect(current.mediaQuery).toEqual(superXLQuery);
+
+      mediaController.activate('tablet-gt-xs');
+      expect(current.mqAlias).toBe('tablet-gt-xs');
+      expect(current.mediaQuery).toBe(gtXsMediaQuery);
+
+      subscription.unsubscribe();
+      if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/media-observer/media-observer.spec.ts]: Do we need this here?

File: src/lib/core/media-observer/media-observer.spec.ts [hunk 6]
```diff
@@ -192,117 +200,146 @@ describe('media-observer', () => {
...
-
-         } finally {
-           matchMedia.clearAll();
-            subscription.unsubscribe();
-         }
-
-       }));
-   });
+        } finally {
+          if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/media-observer/media-observer.spec.ts]: Do we need this here?

File: src/lib/core/media-observer/media-observer.spec.ts [hunk 7]
```diff
@@ -192,117 +200,146 @@ describe('media-observer', () => {
...
+        mediaController.activate('print');
+        expect(current.mqAlias).toBe('sm');
+        expect(current.mediaQuery).toEqual(smMediaQuery);
+
+        mediaController.activate('xl');
+        expect(current.mqAlias).toBe('xl');
+
+      } finally {
+        subscription.unsubscribe();
+        if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/core/media-observer/media-observer.spec.ts]: Do we need this here?

File: src/lib/extended/show-hide/show.spec.ts [hunk 8]
```diff
@@ -349,7 +357,9 @@ describe('show directive', () => {
       });
     });
     afterEach(() => {
-      matchMedia.clearAll();
+      if (mediaController) {
```
Comment (CaerusKaru) [inline src/lib/extended/show-hide/show.spec.ts]: Do we need this here?

**Pull request merged**